src/shogun/machine/Machine.h

```diff
@@ -264,11 +266,33 @@ class Machine : public StoppableSGObject
 			return false;
 		}
 
+		virtual bool train_machine(
```

Comment:
woa isnt this method purely virtual?

Reply:
leaving that part of refactor for when we merge with develop branch